src/pages/core/architecture/actor-model.mdx

```diff
@@ -0,0 +1,55 @@
+---
+tags: ["architecture", "core"]
+---
+
+import Image from "next/image";
+import { Callout } from "nextra/components";
+
+# Actor Model
+
+CosmWasm, at its core, is built around the actor model. This prevents some
+common pitfalls which, for example, Ethereum smart contracts had to fix over the
+years. The issues mostly revolved around reentrancy issues where we could, for
+example, call out to another contract before finishing the current execution.
+
+<Callout>
+  If you are already familiar with the actor model through, for example,
+  programming in languages such as Erlang or Elixir, you can skip this page.
+</Callout>
+
+While the term "actor model" sounds fancy, it is actually quite simple. Each
+contract is an "actor" and _plays_ a role in the system (pun intended).
+
+Each actor has its own state and can only be interacted with via messages. This
+means a contract can only interact with the outside world via messages and only
+manipulate its own state.
+
+Picture two people living in two houses, each house has a mailbox. If person A
+wants to talk to person B, they write a letter and put it in person B's mailbox.
+Person B will read the letter and write a response, putting it in person A's
+mailbox.
+
+No direct interaction is happening here, nobody visits each other's house which
+would allow them to rummage around in their belongings (state). They simply
+exchange messages and put everything the other side needs to know into a
+message.
+
+<Image
+  src="/actor-model.svg"
+  alt="Actor model"
+  style={{ marginTop: "1em", marginBottom: "1em" }}
+  width={1920}
+  height={1080}
+/>
```

Comment:
This graphic is great, just slightly hard to see in light mode:

![image](https://github.com/CosmWasm/docs/assets/360248/acbfdf4f-fdda-4986-949b-57f27ed6ce2f)

Reply:
I'm gonna look into whether there's some CSS trickery where I can adjust the colours in light mode

Reply:
Okay, thanks to SVGs being overengineered and supporting CSS media queries, it should work now.

Reply:
Okay, I'm impressed :grin: I would have probably added a background, but this is nicer